File: frontend/src/types/report.ts
export type DisasterType = "flood" | "landslide" | "fire";
// PRD v2.3 menambah kritis; backend/app/schemas/report.py perlu disinkronkan sebelum C1.
export type Severity = "rendah" | "sedang" | "tinggi" | "kritis";
export type ReportStatus = "draft" | "active" | "disputed_hidden";
export type LocationSource = "device" | "map" | "demo";
export type HelpVoteValue = "seen" | "not_seen";
export type HelpStatus = "belum_ada_konfirmasi" | "belum_terlihat" | "terlihat";

export interface FloodDetails {
  water_depth: "<30cm" | "30-100cm" | ">100cm" | null;
  current: "tenang" | "deras" | null;
}

export interface LandslideDetails {
  covered_area_m2: number | null;
}

export interface FireDetails {
  visibility: "jelas" | "terbatas" | "sangat_rendah" | null;
}

export type ReportDetails =
  | ({ type: "flood" } & FloodDetails)
  | ({ type: "landslide" } & LandslideDetails)
  | ({ type: "fire" } & FireDetails);

// Proyeksi publik — sesuai kontrak GET /api/reports di PRD §19 (tanpa photo_path/author_id/koordinat asli).
export interface Report {
  id: string;
  status: ReportStatus;
  type: DisasterType;
  severity: Severity;
  ai_summary: string;
  description: string | null;
  details: ReportDetails | null;
  location_label: string;
  location_source: LocationSource;
  public_lat: number;
  public_lng: number;
  published_at: string | null;
  created_at: string;
  is_demo: boolean;
  help_status: HelpStatus;
  seen_count: number;
  not_seen_count: number;
  false_vote_count: number;
}

File: backend/app/schemas/report.py
from datetime import datetime
from enum import Enum
from typing import Any

from pydantic import BaseModel


class DisasterType(str, Enum):
    flood = "flood"
    landslide = "landslide"
    fire = "fire"


class Severity(str, Enum):
    """4 tingkat keparahan, tiap tingkat punya radius peringatan sendiri (PRD §9.2):
    rendah tidak pernah memicu peringatan; sedang/tinggi/kritis makin luas radiusnya.
    `kritis` = bahaya skala luas/regional (bukan cuma satu titik parah), beda dari
    `tinggi` yang tetap SEGERA tapi cakupannya lebih lokal — lihat app/services/model.py."""

    rendah = "rendah"
    sedang = "sedang"
    tinggi = "tinggi"
    kritis = "kritis"


class ReportStatus(str, Enum):
    draft = "draft"
    active = "active"
    disputed_hidden = "disputed_hidden"


class ResponderStatus(str, Enum):
    pending = "PENDING"
    accepted = "ACCEPTED"


class LocationSource(str, Enum):
    device = "device"
    map = "map"
    demo = "demo"


class HelpStatus(str, Enum):
    belum_ada_konfirmasi = "belum_ada_konfirmasi"
    belum_terlihat = "belum_terlihat"
    terlihat = "terlihat"


class ReportOut(BaseModel):
    """Proyeksi publik laporan. Bentuknya kembar dengan frontend/src/types/report.ts —
    jangan ubah salah satu tanpa koordinasi tim."""

    id: str
    status: ReportStatus
    responder_status: ResponderStatus
    type: DisasterType
    severity: Severity
    ai_summary: str
    description: str | None = None
    details: dict[str, Any] | None = None
    location_label: str
    location_source: LocationSource
    public_lat: float
    public_lng: float
    published_at: datetime | None = None
    created_at: datetime
    is_demo: bool
    help_status: HelpStatus
    seen_count: int
    not_seen_count: int
    false_vote_count: int
